testplan/web_ui/testing/src/Common/utils.js

```diff
@@ -174,3 +175,22 @@ export const toPlainObjectIn = (obj, enumerableOnly = false) => {
 export const joinURLComponent = (base, component) => {
   return `${base.replace(/\/+$/, '').trim()}/${component}`;
 };
+
+/**
+ * Parse JSON string to object by JSON5 https://json5.org/
+ * @param {string} data - The json string
+ * @returns {object}
+ */
+export const transformResponse = (data) => {
+  let result = data;
+  if (typeof data === 'string' && data.length) {
+    try {
+      result = JSON5.parse(result);
+    } catch (e) {
+      console.log("Cannot parse to json");
+      console.log(e);
+    }
```

Comment:
Would the string be any useful for the consumer of this transformation. Instead of throwing the exception from here would not this result cryptic exception from elsewhere that need to be tracked back down to here?

Reply:
print console.trace() ?

Reply:
More thinking to let the exception thrown and handle that appropriately where called, as it can be different based on the usecase.